Add commenting enhancements

johngodley opened the pull request:
Enhances the logged out commenting experience by showing an avatar for the email address that is entered. This avatar can be used to open the quick editor, allowing direct modification of the account from the comments page.

It is enabled from the Gravatar Enhanced section of the discussion page:

![image](https://github.com/user-attachments/assets/d3e097b2-72d8-4584-a9d2-8266a3dc16f3)

When enabled it re-arranges the comment form so that email is before name:

![image](https://github.com/user-attachments/assets/ba47a3fe-6839-421d-8517-2aedff283588)

When an email is entered we then show the Gravatar profile inline in the field:

![image](https://github.com/user-attachments/assets/bbfc9264-c09c-4340-92a6-f22d60cb0d12)

If the user hasn't already entered a name and URL in the comment form these will be pre-filled from the Gravatar profile.

The icon will slide into place and briefly pulsate:

https://github.com/user-attachments/assets/b86080ef-aff6-4174-97df-7fcc88ee85b9

Clicking the icon will open up the Gravatar quick editor.

This will require https://github.com/Automattic/gravatar/pull/166 to be included in order to not trigger the hovercard when opening the quick editor.

# Caveats

The code should adjust to most themes but some are problematic. I've confirmed it working with all WordPress themes except Twenty Fifteen, which does it's own input field thing.

# Testing

- `yarn build`
- Go to post as a logged out user and enter an email address
- Confirm an avatar is shown in the email field. Clicking this should open the quick editor

johngodley committed:
Add initial commenting enhancements

johngodley committed:
Improve positioning and animation

johngodley committed:
Dont show hovercard if QE pressed

johngodley committed:
1 decimal place rounding

This makes the positioning a bit better

johngodley committed:
Remove pulsating

johngodley committed:
Use first verified account for URL

johngodley committed:
Dont show when logged in

wellyshen commented on src/comments/index.ts:
```diff
@@ -0,0 +1,190 @@
+}
+
+function adjustGravatarPosition() {
+	const gravatarProfile = document.querySelector( GRAVATAR_CONTAINER ) as HTMLElement;
```
Nit: The class has been added to the `<span/>`:

```suggestion
	const gravatarProfile = document.querySelector( GRAVATAR_CONTAINER ) as HTMLSpanElement;
```

wellyshen commented on src/comments/index.ts:
```diff
@@ -0,0 +1,190 @@
+				adjustGravatarPosition();
+			}
+		}, DEBOUNCE_TIMEOUT );
+	} );
```
Suggestion: I recommend using [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) with [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestAnimationFrame) for improved performance and a smoother UX.

wellyshen commented on src/types.d.ts:
```diff
@@ -53,5 +58,7 @@ declare global {
 		recordTrackEvent: ( name: string, options?: any ) => void;
 	};
 
+	var gravatarEnhancedComments : GravatarEnhancedComments;
```
Nit: Code formatting:

```suggestion
	var gravatarEnhancedComments: GravatarEnhancedComments;
```

wellyshen reviewed (commented):
This feature is COOL! I have found some issues during my testing and left some comments, please take a look 👀

- The hovercard and "Website" URL have been updated after I changed the email address:

https://github.com/user-attachments/assets/c3c31533-2fb1-4450-801f-1ed68c5e39a6

- The "Show Gravatar in the comment form" option can't be disabled

https://github.com/user-attachments/assets/9ec925a3-0b7f-49e8-9d9f-ddb47211be88

johngodley commented:
> The "Show Gravatar in the comment form" option can't be disabled

Ha, it's a feature! 😄

wellyshen commented:
> Ha, it's a feature! 😄

That's true 😂

johngodley committed:
Save the comment option

johngodley committed:
Use resize observer

johngodley commented:
I've added the suggested changes and fixed the option saving.

Regarding the problem with the old hovercard being shown after changing the email, this seems to be a problem in the hovercard library which I've put a fix in for at https://github.com/Automattic/gravatar/pull/168

wellyshen reviewed (approved):
Thank you for addressing the feedback. The feature works like a charm 💯

johngodley committed:
Fix rebase

johngodley committed:
Fix comment saving

johngodley committed:
Use latest hovercards
